kernel 4.20

hello after i update my kernel to kernel 4.20 i cant build 88x2bu drivers. in my logs i have this:

DKMS make.log for rtl88x2bu-5.3.1 for kernel 4.20.0-pf1 (x86_64)
śro, 26 gru 2018, 17:02:39 CET
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/4.20.0-pf1/build M=/var/lib/dkms/rtl88x2bu/5.3.1/build  modules
make[1]: Wejście do katalogu '/usr/lib/modules/4.20.0-pf1/build'
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_cmd.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_security.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_debug.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_io.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_ioctl_query.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_ioctl_set.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_ieee80211.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_mlme.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_mlme_ext.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_mi.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_wlan_util.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_vht.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_pwrctrl.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_rf.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_recv.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_sta_mgt.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_ap.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/mesh/rtw_mesh.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/mesh/rtw_mesh_pathtbl.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/mesh/rtw_mesh_hwmp.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_xmit.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_p2p.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_rson.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_tdls.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_br_ext.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_iol.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_sreset.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_btcoex_wifionly.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_btcoex.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_beamforming.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_odm.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_rm.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/rtw_rm_fsm.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/core/efuse/rtw_efuse.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/osdep_service.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/os_intfs.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/usb_intf.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/usb_ops_linux.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/ioctl_linux.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/xmit_linux.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/mlme_linux.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/recv_linux.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/ioctl_cfg80211.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/rtw_cfgvendor.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/wifi_regd.o
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_cfg80211.c: In function ‘rtw_get_systime_us’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_cfg80211.c:691:2: error: implicit declaration of function ‘get_monotonic_boottime’; did you mean ‘getboottime’? [-Werror=implicit-function-declaration]
  get_monotonic_boottime(&ts);
  ^~~~~~~~~~~~~~~~~~~~~~
  getboottime
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/rtw_android.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/rtw_proc.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/rtw_rhashtable.o
  CC [M]  /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/ioctl_mp.o
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:291: /var/lib/dkms/rtl88x2bu/5.3.1/build/os_dep/linux/ioctl_cfg80211.o] Błąd 1
make[2]: *** Oczekiwanie na niezakończone zadania....
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_write_reg’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:46:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length + 1];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_read_reg’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:132:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length + 1];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_write_rf’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:266:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_read_rf’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:308:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_rate’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:417:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_channel’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:467:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_ch_offset’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:496:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_bandwidth’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:524:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_txpower_index’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:556:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length + 1];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_txpower’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:585:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_ant_tx’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:623:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_ant_rx’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:671:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_disable_bt_coexist’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:851:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8 input[wrqu->data.length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_arx’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:889:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_pwrtrk’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1068:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_psd’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1106:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length + 1];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_dump’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1221:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_phypara’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1247:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_SetRFPath’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1273:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_switch_rf_path’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1322:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_QueryDrv’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1371:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  char input[wrqu->data.length];
  ^~~~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_PwrCtlDM’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:1400:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->length];
  ^~
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c: In function ‘rtw_mp_hwtx’:
/usr/src/rtl88x2bu-5.3.1/os_dep/linux/ioctl_mp.c:2050:2: warning: ISO C90 forbids variable length array ‘input’ [-Wvla]
  u8  input[wrqu->data.length];
  ^~
make[1]: *** [Makefile:1563: _module_/var/lib/dkms/rtl88x2bu/5.3.1/build] Błąd 2
make[1]: Opuszczenie katalogu '/usr/lib/modules/4.20.0-pf1/build'
make: *** [Makefile:1986: modules] Błąd 2

--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -684,6 +684,7 @@ static int rtw_cfg80211_sync_iftype(_adapter *adapter)
 	return _SUCCESS;
 }
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 20, 0))
 static u64 rtw_get_systime_us(void)
 {
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 39))
@@ -696,6 +697,7 @@ static u64 rtw_get_systime_us(void)
 	return ((u64)tv.tv_sec * 1000000) + tv.tv_usec;
 #endif
 }
+#endif
 
 /* Try to remove non target BSS's SR to reduce PBC overlap rate */
 static int rtw_cfg80211_clear_wps_sr_of_non_target_bss(_adapter *padapter, struct wlan_network *pnetwork, struct cfg80211_ssid *req_ssid)
@@ -789,7 +791,11 @@ struct cfg80211_bss *rtw_cfg80211_inform_bss(_adapter *padapter, struct wlan_net
 	if (0)
 		notify_timestamp = le64_to_cpu(*(u64 *)rtw_get_timestampe_from_ie(pnetwork->network.IEs));
 	else
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 20, 0))
+		notify_timestamp = ktime_to_us(ktime_get_boottime());
+#else
 		notify_timestamp = rtw_get_systime_us();
+#endif
 
 	notify_interval = le16_to_cpu(*(u16 *)rtw_get_beacon_interval_from_ie(pnetwork->network.IEs));
 	notify_capability = le16_to_cpu(*(u16 *)rtw_get_capability_from_ie(pnetwork->network.IEs));

--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -80,7 +80,7 @@
 #endif
 
 /*
- * In the current design of Wi-Fi driver, it will return success to the system (e.g. supplicant) 
+ * In the current design of Wi-Fi driver, it will return success to the system (e.g. supplicant)
  * when Wi-Fi driver decides to abort the scan request in the scan flow by default.
  * Defining this flag makes Wi-Fi driver to return -EBUSY to the system if Wi-Fi driver is too busy to do the scan.
  */
@@ -321,7 +321,7 @@ static u8 rtw_chbw_to_cfg80211_chan_def(struct wiphy *wiphy, struct cfg80211_cha
 	if (!chan)
 		goto exit;
 
-	if (bw == CHANNEL_WIDTH_20) 
+	if (bw == CHANNEL_WIDTH_20)
 		chdef->width = ht ? NL80211_CHAN_WIDTH_20 : NL80211_CHAN_WIDTH_20_NOHT;
 	else if (bw == CHANNEL_WIDTH_40)
 		chdef->width = NL80211_CHAN_WIDTH_40;
@@ -355,7 +355,7 @@ static void rtw_get_chbw_from_cfg80211_chan_def(struct cfg80211_chan_def *chdef,
 		rtw_warn_on(1);
 		*ch = 0;
 		return;
-	}		
+	}
 
 	switch (chdef->width) {
 	case NL80211_CHAN_WIDTH_20_NOHT:
@@ -1587,8 +1587,8 @@ static int rtw_cfg80211_set_encryption(struct net_device *dev, struct ieee_param
 						_rtw_memcpy(padapter->securitypriv.dot118021XGrptxmickey[param->u.crypt.idx].skey, &(param->u.crypt.key[16]), 8);
 						_rtw_memcpy(padapter->securitypriv.dot118021XGrprxmickey[param->u.crypt.idx].skey, &(param->u.crypt.key[24]), 8);
 						padapter->securitypriv.binstallGrpkey = _TRUE;
-						if (param->u.crypt.idx < 4) 
-							_rtw_memcpy(padapter->securitypriv.iv_seq[param->u.crypt.idx], param->u.crypt.seq, 8);							
+						if (param->u.crypt.idx < 4)
+							_rtw_memcpy(padapter->securitypriv.iv_seq[param->u.crypt.idx], param->u.crypt.seq, 8);
 						padapter->securitypriv.dot118021XGrpKeyid = param->u.crypt.idx;
 						rtw_set_key(padapter, &padapter->securitypriv, param->u.crypt.idx, 1, _TRUE);
 
@@ -1833,7 +1833,7 @@ static int cfg80211_rtw_get_key(struct wiphy *wiphy, struct net_device *ndev
 		#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
 		|| (MLME_IS_STA(adapter) && !pairwise)
 		#endif
-	) {	
+	) {
 		/* WEP key, TX GTK/IGTK, RX GTK/IGTK(for STA mode) */
 		if (is_wep_enc(sec->dot118021XGrpPrivacy)) {
 			if (keyid >= WEP_KEYS)
@@ -1922,7 +1922,7 @@ static int cfg80211_rtw_get_key(struct wiphy *wiphy, struct net_device *ndev
 	}
 
 	ret = 0;
-	
+
 exit:
 	RTW_INFO(FUNC_NDEV_FMT
 		GET_KEY_PARAM_FMT_S
@@ -5452,7 +5452,7 @@ static int	cfg80211_rtw_dump_station(struct wiphy *wiphy, struct net_device *nde
 	else
 		_rtw_memcpy(mac, plink->addr, ETH_ALEN);
 	#endif
-	
+
 	sinfo->filled = 0;
 
 	if (psta) {
@@ -7400,7 +7400,7 @@ void dump_mesh_config(void *sel, const struct mesh_config *conf)
 	RTW_PRINT_SEL(sel, "path_refresh_time:%u\n", conf->path_refresh_time);
 	RTW_PRINT_SEL(sel, "min_discovery_timeout:%u\n", conf->min_discovery_timeout);
 	RTW_PRINT_SEL(sel, "dot11MeshHWMPactivePathTimeout:%u\n", conf->dot11MeshHWMPactivePathTimeout);
-	RTW_PRINT_SEL(sel, "dot11MeshHWMPpreqMinInterval:%u\n", conf->dot11MeshHWMPpreqMinInterval);	
+	RTW_PRINT_SEL(sel, "dot11MeshHWMPpreqMinInterval:%u\n", conf->dot11MeshHWMPpreqMinInterval);
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0))
 	RTW_PRINT_SEL(sel, "dot11MeshHWMPperrMinInterval:%u\n", conf->dot11MeshHWMPperrMinInterval);
 #endif
@@ -7415,11 +7415,11 @@ void dump_mesh_config(void *sel, const struct mesh_config *conf)
 	RTW_PRINT_SEL(sel, "dot11MeshForwarding:%d\n", conf->dot11MeshForwarding);
 	RTW_PRINT_SEL(sel, "rssi_threshold:%d\n", conf->rssi_threshold);
 #endif
-	
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 5, 0))
 	RTW_PRINT_SEL(sel, "ht_opmode:0x%04x\n", conf->ht_opmode);
 #endif
-	
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
 	RTW_PRINT_SEL(sel, "dot11MeshHWMPactivePathToRootTimeout:%u\n", conf->dot11MeshHWMPactivePathToRootTimeout);
 	RTW_PRINT_SEL(sel, "dot11MeshHWMProotInterval:%u\n", conf->dot11MeshHWMProotInterval);
@@ -7430,7 +7430,7 @@ void dump_mesh_config(void *sel, const struct mesh_config *conf)
 	RTW_PRINT_SEL(sel, "power_mode:%s\n", nl80211_mesh_power_mode_str(conf->power_mode));
 	RTW_PRINT_SEL(sel, "dot11MeshAwakeWindowDuration:%u\n", conf->dot11MeshAwakeWindowDuration);
 #endif
-	
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0))
 	RTW_PRINT_SEL(sel, "plink_timeout:%u\n", conf->plink_timeout);
 #endif
@@ -7566,14 +7566,14 @@ static void rtw_cfg80211_mesh_cfg_set(_adapter *adapter, const struct mesh_confi
 	if (chk_mesh_attr(NL80211_MESHCONF_HT_OPMODE, mask));
 #endif
 #endif
-	
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
 	if (chk_mesh_attr(NL80211_MESHCONF_HWMP_PATH_TO_ROOT_TIMEOUT, mask))
 		mcfg->dot11MeshHWMPactivePathToRootTimeout = conf->dot11MeshHWMPactivePathToRootTimeout;
 	if (chk_mesh_attr(NL80211_MESHCONF_HWMP_ROOT_INTERVAL, mask))
 		mcfg->dot11MeshHWMProotInterval = conf->dot11MeshHWMProotInterval;
 	if (chk_mesh_attr(NL80211_MESHCONF_HWMP_CONFIRMATION_INTERVAL, mask))
-		mcfg->dot11MeshHWMPconfirmationInterval = conf->dot11MeshHWMPconfirmationInterval;	
+		mcfg->dot11MeshHWMPconfirmationInterval = conf->dot11MeshHWMPconfirmationInterval;
 #endif
 
 #if 0 /* TBD */
@@ -7631,7 +7631,7 @@ u8 *rtw_cfg80211_construct_mesh_beacon_ies(struct wiphy *wiphy, _adapter *adapte
 #endif
 	if (!ch)
 		goto exit;
-	
+
 #if defined(CONFIG_80211AC_VHT) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
 	vht = ht && ch > 14 && bw >= CHANNEL_WIDTH_80; /* VHT40/VHT20? */
 #endif
@@ -7956,7 +7956,7 @@ static int cfg80211_rtw_join_mesh(struct wiphy *wiphy, struct net_device *dev,
 		ret = -EINVAL;
 		goto exit;
 	}
-	
+
 	rtw_mesh_work(&adapter->mesh_work);
 
 exit:
@@ -8037,7 +8037,7 @@ static int cfg80211_rtw_del_mpath(struct wiphy *wiphy, struct net_device *dev
 		}
 	} else {
 		rtw_mesh_path_flush_by_iface(adapter);
-	}	
+	}
 
 exit:
 	return ret;
@@ -8333,13 +8333,13 @@ int	cfg80211_rtw_resume(struct wiphy *wiphy) {
 		//rtw_sitesurvey_cmd(padapter, NULL);
 		rtw_sitesurvey_cmd(padapter, &parm);
 		_exit_critical_bh(&pmlmepriv->lock, &irqL);
-		
+
 		for (PNOWakeupScanWaitCnt = 0; PNOWakeupScanWaitCnt < 10; PNOWakeupScanWaitCnt++) {
 			if(check_fwstate(pmlmepriv, _FW_UNDER_SURVEY) == _FALSE)
 				break;
 			rtw_msleep_os(1000);
 		}
-		
+
 		_enter_critical_bh(&pmlmepriv->lock, &irqL);
 		cfg80211_sched_scan_results(padapter->rtw_wdev->wiphy);
 		_exit_critical_bh(&pmlmepriv->lock, &irqL);
@@ -8347,7 +8347,7 @@ int	cfg80211_rtw_resume(struct wiphy *wiphy) {
 	}
 	RTW_DBG("<== %s\n",__func__);
 	return 0;
-	
+
 }
 #endif /* CONFIG_PNO_SUPPORT */
 
@@ -9235,7 +9235,7 @@ int rtw_hostapd_acs_dump_survey(struct wiphy *wiphy, struct net_device *netdev, 
 #elif defined(CONFIG_RTW_ACS) && defined(CONFIG_BACKGROUND_NOISE_MONITOR)
 	rtw_cfg80211_set_survey_info_with_clm(padapter, idx, info);
 #else
-	RTW_ERR("%s: unknown acs operation!\n", __func__); 
+	RTW_ERR("%s: unknown acs operation!\n", __func__);
 #endif
 
 	return ret;

